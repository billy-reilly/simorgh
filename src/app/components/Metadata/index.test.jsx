import React from 'react';
import Metadata from './index';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';

const metadataSnapshotTest = (
  testDescription,
  isAmp,
  alternateLinks,
  ampLink,
  articleAuthor,
  articleSection,
  appleTouchIcon,
  brandName,
  canonicalLink,
  defaultImage,
  defaultImageAltText,
  description,
  facebookAdmin,
  facebookAppID,
  lang,
  locale,
  metaTags,
  themeColor,
  timeLastPublished,
  timeFirstPublished,
  title,
  twitterCreator,
  twitterSite,
) =>
  describe(testDescription, () => {
    const metadataProps = {
      isAmp,
      alternateLinks,
      ampLink,
      articleAuthor,
      articleSection,
      appleTouchIcon,
      brandName,
      canonicalLink,
      defaultImage,
      defaultImageAltText,
      description,
      facebookAdmin,
      facebookAppID,
      lang,
      locale,
      metaTags,
      themeColor,
      timeLastPublished,
      timeFirstPublished,
      title,
      twitterCreator,
      twitterSite,
    };

    shouldShallowMatchSnapshot(
      'should render correctly',
      <Metadata {...metadataProps} />,
    );
  });

describe('Metadata', () => {
  metadataSnapshotTest(
    'News article',
    false,
    [
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o',
        hrefLang: 'x-default',
      },
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o',
        hrefLang: 'en',
      },
      {
        href: 'https://www.bbc.co.uk/news/articles/c0000000001o',
        hrefLang: 'en-gb',
      },
    ],
    'https://www.bbc.com/news/articles/c0000000001o.amp',
    'BBC News',
    null,
    'https://foo.com/static/news/image.png',
    'BBC News',
    'https://www.bbc.com/news/articles/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    101010,
    202020,
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    '#B80000',
    1539188371344,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );

  metadataSnapshotTest(
    'News AMP article',
    true,
    [
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o.amp',
        hrefLang: 'x-default',
      },
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o.amp',
        hrefLang: 'en',
      },
      {
        href: 'https://www.bbc.co.uk/news/articles/c0000000001o.amp',
        hrefLang: 'en-gb',
      },
    ],
    'https://www.bbc.com/news/articles/c0000000001o.amp',
    'BBC News',
    null,
    'https://foo.com/static/news/image.png',
    'BBC News',
    'https://www.bbc.com/news/articles/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    101010,
    202020,
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    '#B80000',
    1539188371344,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );

  metadataSnapshotTest(
    'Persian article',
    false,
    [],
    'https://www.bbc.com/persian/articles/c4vlle3q337o.amp',
    'BBC News فارسی',
    null,
    'https://foo.com/static/persian/image.png',
    'BBC News فارسی',
    'https://www.bbc.com/persian/articles/c4vlle3q337o',
    'https://www.bbc.com/persian/image.png',
    'BBC News فارسی',
    'This is a description',
    101010,
    202020,
    'fa',
    'fa',
    ['tagA', 'tagB'],
    '#B80000',
    1539188371344,
    1514811600000,
    'پهپادی که برایتان قهوه می‌آورد',
    '@bbcpersian',
    '@bbcpersian',
  );

  metadataSnapshotTest(
    'Persian AMP article',
    true,
    [],
    'https://www.bbc.com/persian/articles/c4vlle3q337o.amp',
    'BBC News فارسی',
    null,
    'https://foo.com/static/persian/image.png',
    'BBC News فارسی',
    'https://www.bbc.com/persian/articles/c4vlle3q337o',
    'https://www.bbc.com/persian/image.png',
    'BBC News فارسی',
    'This is a description',
    101010,
    202020,
    'fa',
    'fa',
    ['tagA', 'tagB'],
    '#B80000',
    1539188371344,
    1514811600000,
    'پهپادی که برایتان قهوه می‌آورد',
    '@bbcpersian',
    '@bbcpersian',
  );

  metadataSnapshotTest(
    'articleSection is not null',
    false,
    [
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o',
        hrefLang: 'x-default',
      },
      {
        href: 'https://www.bbc.com/news/articles/c0000000001o',
        hrefLang: 'en',
      },
      {
        href: 'https://www.bbc.co.uk/news/articles/c0000000001o',
        hrefLang: 'en-gb',
      },
    ],
    'https://www.bbc.com/news/articles/c0000000001o.amp',
    'BBC News',
    'Politics',
    'https://foo.com/static/news/image.png',
    'BBC News',
    'https://www.bbc.com/news/articles/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    101010,
    202020,
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    '#B80000',
    1539188371344,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );
});
