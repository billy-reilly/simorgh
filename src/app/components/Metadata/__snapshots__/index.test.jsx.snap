// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata News AMP article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "amp": "",
      "lang": "en-GB",
    }
  }
>
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="noodp,noydir"
    name="robots"
  />
  <meta
    content="#B80000"
    name="theme-color"
  />
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
     - 
    BBC News
  </title>
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    rel="canonical"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o.amp"
    hrefLang="x-default"
    rel="alternate"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o.amp"
    hrefLang="en"
    rel="alternate"
  />
  <link
    href="https://www.bbc.co.uk/news/articles/c0000000001o.amp"
    hrefLang="en-gb"
    rel="alternate"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1539188371344}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="description"
  />
  <meta
    content={101010}
    name="fb:admins"
  />
  <meta
    content={202020}
    name="fb:app_id"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
  <link
    href="/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  <link
    href="https://foo.com/static/news/image.png"
    rel="apple-touch-icon"
  />
</HelmetWrapper>
`;

exports[`Metadata News article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en-GB",
    }
  }
>
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="noodp,noydir"
    name="robots"
  />
  <meta
    content="#B80000"
    name="theme-color"
  />
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
     - 
    BBC News
  </title>
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    rel="canonical"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    hrefLang="x-default"
    rel="alternate"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    hrefLang="en"
    rel="alternate"
  />
  <link
    href="https://www.bbc.co.uk/news/articles/c0000000001o"
    hrefLang="en-gb"
    rel="alternate"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o.amp"
    rel="amphtml"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1539188371344}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="description"
  />
  <meta
    content={101010}
    name="fb:admins"
  />
  <meta
    content={202020}
    name="fb:app_id"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
  <link
    href="/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  <link
    href="https://foo.com/static/news/image.png"
    rel="apple-touch-icon"
  />
</HelmetWrapper>
`;

exports[`Metadata Persian AMP article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "amp": "",
      "lang": "fa",
    }
  }
>
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="noodp,noydir"
    name="robots"
  />
  <meta
    content="#B80000"
    name="theme-color"
  />
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    پهپادی که برایتان قهوه می‌آورد
     - 
    BBC News فارسی
  </title>
  <link
    href="https://www.bbc.com/persian/articles/c4vlle3q337o"
    rel="canonical"
  />
  <meta
    content="BBC News فارسی"
    name="article:author"
  />
  <meta
    content={1539188371344}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="description"
  />
  <meta
    content={101010}
    name="fb:admins"
  />
  <meta
    content={202020}
    name="fb:app_id"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="og:image"
  />
  <meta
    content="BBC News فارسی"
    name="og:image:alt"
  />
  <meta
    content="fa"
    name="og:locale"
  />
  <meta
    content="BBC News فارسی"
    name="og:site_name"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/persian/articles/c4vlle3q337o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@bbcpersian"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News فارسی"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@bbcpersian"
    name="twitter:site"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="twitter:title"
  />
  <link
    href="/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  <link
    href="https://foo.com/static/persian/image.png"
    rel="apple-touch-icon"
  />
</HelmetWrapper>
`;

exports[`Metadata Persian article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "fa",
    }
  }
>
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="noodp,noydir"
    name="robots"
  />
  <meta
    content="#B80000"
    name="theme-color"
  />
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    پهپادی که برایتان قهوه می‌آورد
     - 
    BBC News فارسی
  </title>
  <link
    href="https://www.bbc.com/persian/articles/c4vlle3q337o"
    rel="canonical"
  />
  <link
    href="https://www.bbc.com/persian/articles/c4vlle3q337o.amp"
    rel="amphtml"
  />
  <meta
    content="BBC News فارسی"
    name="article:author"
  />
  <meta
    content={1539188371344}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="description"
  />
  <meta
    content={101010}
    name="fb:admins"
  />
  <meta
    content={202020}
    name="fb:app_id"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="og:image"
  />
  <meta
    content="BBC News فارسی"
    name="og:image:alt"
  />
  <meta
    content="fa"
    name="og:locale"
  />
  <meta
    content="BBC News فارسی"
    name="og:site_name"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/persian/articles/c4vlle3q337o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@bbcpersian"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News فارسی"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@bbcpersian"
    name="twitter:site"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="twitter:title"
  />
  <link
    href="/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  <link
    href="https://foo.com/static/persian/image.png"
    rel="apple-touch-icon"
  />
</HelmetWrapper>
`;

exports[`Metadata articleSection is not null should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en-GB",
    }
  }
>
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="noodp,noydir"
    name="robots"
  />
  <meta
    content="#B80000"
    name="theme-color"
  />
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
     - 
    BBC News
  </title>
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    rel="canonical"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    hrefLang="x-default"
    rel="alternate"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    hrefLang="en"
    rel="alternate"
  />
  <link
    href="https://www.bbc.co.uk/news/articles/c0000000001o"
    hrefLang="en-gb"
    rel="alternate"
  />
  <link
    href="https://www.bbc.com/news/articles/c0000000001o.amp"
    rel="amphtml"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1539188371344}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="Politics"
    name="article:section"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="description"
  />
  <meta
    content={101010}
    name="fb:admins"
  />
  <meta
    content={202020}
    name="fb:app_id"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
  <link
    href="/favicon.ico"
    rel="shortcut icon"
    type="image/x-icon"
  />
  <link
    href="https://foo.com/static/news/image.png"
    rel="apple-touch-icon"
  />
</HelmetWrapper>
`;
